BezierKit/Library/Polynomial.swift

```diff
@@ -11,51 +11,52 @@ import CoreGraphics
 #endif
 import Foundation
 
-internal protocol BernsteinPolynomial: Equatable {
-    func f(_ x: Double) -> Double
+public protocol BernsteinPolynomial: Equatable {
+    func f(_ x: CGFloat) -> CGFloat
     var order: Int { get }
-    var coefficients: [Double] { get }
-//    var last: Double { get }
-//    var first: Double { get }
-//    func enumerated(block: (Int, Double) -> Void)
+    var coefficients: [CGFloat] { get }
+//    var last: CGFloat { get }
+//    var first: CGFloat { get }
+//    func enumerated(block: (Int, CGFloat) -> Void)
     associatedtype Difference: BernsteinPolynomial
-    func difference(a1: Double, a2: Double) -> Difference
-    func reduce(a1: Double, a2: Double) -> Double
+    func difference(a1: CGFloat, a2: CGFloat) -> Difference
```

Comment:
Can we make `difference` and `reduce` internal? It's not clear at all what they even mean. Maybe we add a separate protocol to keep them internal?

Reply:
not right now, no.